app/lib/actions/pulp/repository/remove_units.rb

```diff
@@ -0,0 +1,28 @@
+module Actions
+  module Pulp
+    module Repository
+      class RemoveUnits < Pulp::AbstractAsyncTask
+        input_format do
+          param :repo_id
+          param :contents
+        end
+
+        def invoke_external_task
+          repo = ::Katello::Repository.find(input[:repo_id])
+          if(input[:contents])
+            ::SmartProxy.pulp_master.content_service(::Katello::RepositoryTypeManager.find(repo.content_type).content_types.first).remove(repo.pulp_id, input[:contents])
```

Comment:
`content_types.first` should return content_type based on priority. Checked the registration and all user removable content types are either single types or priority 1.